core/src/main/java/org/javarosa/core/api/NameHasher.java

```diff
@@ -0,0 +1,22 @@
+package org.javarosa.core.api;
+
+import org.javarosa.core.util.externalizable.DefaultHasher;
+
+
+/**
+ * Most simple possible "hasher" - just gets the byte representation of the hash name
+ *
+ * @author wspride
+ */
+public class NameHasher extends DefaultHasher {
+
+    public final static int CLASS_HASH_SIZE = 32;
+
+    public byte[] doHash(Class c){
```

Comment:
Since J2ME doesn't support override annotations, please add a comment saying that you are overriding the DefaultHasher.doHash